### PR TITLE
Bugfix/unit progress save

### LINKED ISSUE #269: file upload UI/UX
# User Story

### As a: 

Teacher

### I want:

a simple intuitive way to upload files

### so that:

- it works distraction free and gives feedback on successful file upload
- it accepts a bunch of popular file formats

## Acceptance criteria:

- [x] usability review
- [x] restyling

## Additional ideas:

- [x] a function to import an external .csv file with infos eg. about attending course members (already done with #41)

### LINKED ISSUE #379: can't cancel fileuploads
If you hit the cancel-button during a file-Upload the upload does not cancel and will be continued in the background.

### LINKED ISSUE #380: 🐛 BUG: can't restart a fileupload
when a fileupload fails you can't restart it by clicking upload again.
In order to restart it (or upload something else) you currently need to close and reopen the unit.

### LINKED ISSUE #425: Refactor unit and progress
# User Story

### As a: 
developer

### I want:
a nice clean way for handling units and progress

### so that:
new unit and progress types can be added more easily

## Acceptance criteria:
- [x] unified controller for units
- [x] unified controller for progress
- [x] migration script for old database schema 

## Additional info:
#269  will be solved together with this

------
_Please tag this issue if you are sure to which tag(s) it belongs._

### LINKED ISSUE #452: As a student, I can't visit the courses I am enrolled in
I am logged in as a student and I am successfully enrolled in some courses. Currently, there is no way for me to enter the course page:

capture

### PR DESCRIPTION
**Pull request by grosspersky**
## Description:

This PR provides a major refactoring for the whole unit and progress workflow in the database. Because of this major refactoring the database needs migration. This PR provides a basic migration handler for this case.

To execute the currently implemented migrations run the following commands:
``npm run migrate -- --up unit``
``npm run migrate -- --up taskUnit videoUnit``

It is important to note that the unit migration must be executed before the other, more specialized unit migrations run, because the other migrations already depent on the new schema used in the general unit migration script.

Closes #269 
Closes #379 
Closes #425 
Closes #452 

## Improvements
- New reuseable file upload form
- Much cleaner way for saving units and corresponding progress if a unit is progressable.
- Overall unified implementation for units and progress

## Known Issues:
- Migration scripts needs to be run.

------
_Remember to prefix your PR-Title with `⚠ WIP: ` if you still work on it.
If you have reached a final state, remove the prefix._

### CODE REVIEW

File: api/src/server.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export class Server {
   }
 
   start() {
+    mongoose.set('debug', true);
```
Comment (lukaskorte) [inline api/src/server.ts]: This should probably be removed

File: app/webFrontend/src/app/course/course.component.ts [hunk 2]
```diff
@@ -80,24 +80,12 @@ export class CourseComponent {
...
-    return ( course.teachers.filter(teacher => teacher._id === this.userService.user._id).length)
-  }
-
   isMemberOfCourse(course: ICourse) {
     const user = this.userService.user;
+    /*
     return this.userService.isStudent() &&
       course.students.filter(obj => obj._id === user._id).length > 0;
+      */
+    return true;
```
Comment (lukaskorte) [inline app/webFrontend/src/app/course/course.component.ts]: ?

**Comment by grosspersky** [comment]
How would we do this automatic migration. There is nothing implemented to execute just all available migration. Also how should travis know which database needs to be updated?

File: app/webFrontend/src/app/course/course.component.ts (hunk 2 above)
Comment (grosspersky) [reply app/webFrontend/src/app/course/course.component.ts]: Noticed the error David reported in #452 and "fixed" it. 😅 

Now fixed it the correct way.

**Comment by lukaskorte** [comment]
Ok, that will needed to be integrated to the deploy script on our server. @khase can you do that?

**Pull request merged**